### PR TITLE
[camera][android] Support `ratio` prop

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Adds support for setting the aspect ratio on android.

# How
Added `ratio` prop. CameraX supports `4:3` and `16:9`. Based on this is will select this ratio or the closest available depending on the device.

# Test Plan
bare-expo

### CODE REVIEW

File: packages/expo-camera/src/Camera.types.ts [hunk 1]
```diff
@@ -387,6 +371,30 @@ export type CameraProps = ViewProps & {
    * @platform ios
    */
   responsiveOrientationWhenOrientationLocked?: boolean;
+  /**
+   * A string representing aspect ratio of the preview, eg. `4:3`, `16:9`.
```
Comment (amandeepmittal) [inline packages/expo-camera/src/Camera.types.ts:375]:
```suggestion
   * A string representing the aspect ratio of the preview. For example, `4:3` and `16:9`.
```

File: packages/expo-camera/src/Camera.types.ts [hunk 2]
```diff
@@ -387,6 +371,30 @@ export type CameraProps = ViewProps & {
...
+   * @default 4:3
+   * @platform android
+   */
+  ratio?: CameraRatio;
+  /**
+   * Callback invoked when camera preview has been set.
+   */
+  onCameraReady?: () => void;
+  /**
+   * Callback invoked when camera preview could not been started.
```
Comment (amandeepmittal) [inline packages/expo-camera/src/Camera.types.ts:385]:
```suggestion
   * Callback invoked when camera preview could not start.
```

File: packages/expo-camera/src/Camera.types.ts [hunk 3]
```diff
@@ -387,6 +371,30 @@ export type CameraProps = ViewProps & {
...
+   * @param event Error object that contains a `message`.
+   */
+  onMountError?: (event: CameraMountError) => void;
+  /**
+   * Callback that is invoked when a barcode has been successfully scanned. The callback is provided with
+   * an object of the [`BarcodeScanningResult`](#barcodescanningresult) shape, where the `type`
+   * refers to the barcode type that was scanned and the `data` is the information encoded in the barcode
+   * (in this case of QR codes, this is often a URL). See [`BarcodeType`](#barcodetype) for supported values.
+   * for supported values.
```
Comment (amandeepmittal) [inline packages/expo-camera/src/Camera.types.ts:394]:
```suggestion
   * Callback that is invoked when a barcode has been successfully scanned. The callback is provided with
   * an object of the [`BarcodeScanningResult`](#barcodescanningresult) shape, where the `type`
   * refers to the barcode type that was scanned, and the `data` is the information encoded in the barcode
   * (in this case of QR codes, this is often a URL). See [`BarcodeType`](#barcodetype) for supported values.
   * for supported values.
```

**Review by amandeepmittal — COMMENTED** [review]
Can we also add the instruction back to `takePictureAsync()` method which was removed from this PR: https://github.com/expo/expo/pull/29802?

File: packages/expo-camera/src/Camera.types.ts (hunk 3 above)
Comment (alanjhughes) [reply packages/expo-camera/src/Camera.types.ts:394]: @amandeepmittal no, it's not necessary here. In the old package it needed to be explicitly set or you would get weird behaviour. The default now is the `CameraX` default `4:3`.

**Comment by alanjhughes** [comment]
@amandeepmittal no, it's not necessary here. In the old package it needed to be explicitly set or you would get weird behaviour. The default now is the CameraX default 4:3.

**Review by amandeepmittal — APPROVED** [review]

**Comment by timheilman** [comment]
The workaround for [the android camera stretching problem](https://expo.canny.io/feature-requests/p/manage-camera-stretching-problem-in-android-device) also depends upon [getSupportedRatiosAsync](https://docs.expo.dev/versions/v50.0.0/sdk/camera/#getsupportedratiosasync).  Should that method be provided too?

**Comment by DudeFactory** [comment]
Thanks for changes. But I don't understand where these changes? I can't find it in 15.0.12. How get this changes? Or how long to wait when they are published?

**Comment by alanjhughes** [comment]
@timheilman there is no need for that function in the new package. The camera is never stretched or distorted.

**Comment by DudeFactory** [comment]
@alanjhughes before 51 we definitely changed it. We have a popup where we use the camera as a square, and we have a place where we use it as a full screen.  So how does it automatically choose what we need?

**Comment by alanjhughes** [comment]
@DudeFactory the default is 4:3. Not all devices support the same aspect ratios so you request what you want and depending on the device you'll get the closest possible to what you asked for.

**Comment by DanTol-cloud** [comment]
@alanjhughes as I understood that the changes that will allow you to automatically set the ratio have not yet been made. So how now to make ratio was 4:3?

**Comment by alanjhughes** [comment]
@DanTol-cloud its the default

**Comment by DanTol-cloud** [comment]
@alanjhughes but we exactly have a 16:9. So how can i set a 4:3 or in future updates it will be ability to set 4:3?

**Comment by DanTol-cloud** [comment]
cause i try to set ratio prop and it says to me that it is undefined on version: 15.0.14

**Comment by DanTol-cloud** [comment]
> @alanjhughes but we exactly have a 16:9. So how can i set a 4:3 or in future updates it will be ability to set 4:3?

and 16:9 for us is not working

**Comment by alanjhughes** [comment]
@DanTol-cloud this change has not been published yet

**Pull request merged**